### PR TITLE
feat: composables docs

### PR DESCRIPTION
**Pull request by ksted**
## Why:

Closes: [AB#80041](https://dev.azure.com/plentymarkets/0af57df9-8662-4901-bb97-8e88b07ff0c9/_workitems/edit/80041)

## Describe your changes

Adds a VitePress docs site for web app composables.

Open warnings:
```
[warning] CookieGroup, defined in ./cookie.config.ts, is referenced by UseCookieReturn.cookieJson but not included in the documentation.
[warning] useProductRecommended, defined in ./composables/useProductRecommended/types.ts, is referenced by useProductRecommended.useProductRecommended but not included in the documentation.
[warning] CookieGroupFromNuxtConfig, defined in ./cookie.config.ts, is referenced by useCookieBar.useCookieBar.initialCookieJsonFromConfig but not included in the documentation.
```

Open errors (when exporting types in `apps/web/composables/useProductRecommended/index.ts`:
```
composables/useProductRecommended/index.ts:2:1 - error TS2308: Module './useProductRecommended' has already exported a member named 'useProductRecommended'. Consider explicitly re-exporting to resolve the ambiguity.

2 export * from './types';
```

It's important to note that the resulting documentation only considers explicitly exported functions. This means internal functions are excluded. [This is by design.](https://stackoverflow.com/questions/66209551/typedoc-how-do-i-generate-documentation-for-internal-functions#comment117206277_66209551):
```
As far as TypeDoc is concerned, this is not documentable code, there's no way to document these. You could technically write a plugin to do this, but eww.... even more eww than documenting not exported functions...
```

In my opinion, we should handle this by extending the documentation of the exported function and keeping the internal annotations as slim as possible. For example, I feel like all the individual `@example` blocks could be removed. I would like to handle those changes separately, though.

## Checklist before requesting a review
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: apps/web/composables/useProductRecommended/index.ts [hunk 1]
```diff
@@ -1 +1,2 @@
 export * from './useProductRecommended';
+// export * from './types';
```
Comment (maxiroellplenty) [inline apps/web/composables/useProductRecommended/index.ts:2]: Why is this commented?

File: apps/web/docs/index.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    - theme: brand
+      text: Getting started
+      link: /composables/
+    - theme: alt
+      text: API
+      link: /composables/exports
+
+features:
+  - title: Interfaces
+    details: Lorem ipsum dolor sit amet, consectetur adipiscing elit
```
Comment (maxiroellplenty) [inline apps/web/docs/index.md:18]: do we have follow-up tasks to add more details?

File: apps/web/composables/useProductRecommended/index.ts (hunk 1 above)
Comment (ksted) [reply apps/web/composables/useProductRecommended/index.ts:2]: You get the error mentioned in the PR:
```
composables/useProductRecommended/index.ts:2:1 - error TS2308: Module './useProductRecommended' has already exported a member named 'useProductRecommended'. Consider explicitly re-exporting to resolve the ambiguity.

2 export * from './types';
```
I wanted to make a note that it's there, but solve it separately.

File: apps/web/docs/index.md (hunk 2 above)
Comment (ksted) [reply apps/web/docs/index.md:18]: Not yet. I'm not sure if it's worth doing here because we'll probably have a different index for the real site.

Comment (maxiroellplenty) [reply apps/web/docs/index.md:18]: Sounds good to leave it as is.

**Review by maxiroellplenty — APPROVED** [review]

**Pull request merged**